backend/dataall/modules/worksheets/services/worksheet_query_result_service.py

```diff
@@ -0,0 +1,139 @@
+import csv
+import io
+import os
+from datetime import datetime, timedelta
+from typing import TYPE_CHECKING
+
+from openpyxl import Workbook
+
+from dataall.base.aws.s3_client import S3_client
+from dataall.base.db import exceptions
+from dataall.core.environment.services.environment_service import EnvironmentService
+from dataall.modules.worksheets.db.worksheet_models import WorksheetQueryResult
+from dataall.modules.worksheets.db.worksheet_repositories import WorksheetRepository
+from dataall.modules.worksheets.services.worksheet_service import WorksheetService
+
+if TYPE_CHECKING:
+    try:
+        from sqlalchemy.orm import Session
+        from mypy_boto3_s3.client import S3Client
+    except ImportError:
+        print('skipping type checks as stubs are not installed')
+        S3Client = None
+        Session = None
+
+
+class WorksheetQueryResultService:
+    SupportedFormats = {'csv', 'xlsx'}
+
+    @staticmethod
+    def validate_input(data):
+        if not data:
+            raise exceptions.InvalidInput('data', data, 'input is required')
+        if not data.get('athenaQueryId'):
+            raise exceptions.RequiredParameter('athenaQueryId')
+        if not data.get('fileFormat'):
+            raise exceptions.RequiredParameter('fileFormat')
+        if data.get('fileFormat', '').lower() not in WorksheetQueryResultService.SupportedFormats:
+            raise exceptions.InvalidInput(
+                'fileFormat', data.get('fileFormat'), ', '.join(WorksheetQueryResultService.SupportedFormats)
+            )
+
+    @staticmethod
+    def get_output_bucket(session: 'Session', environment_uri: str) -> str:
+        environment = EnvironmentService.get_environment_by_uri(session, environment_uri)
+        bucket = environment.EnvironmentDefaultBucketName
+        return bucket
+
+    @staticmethod
+    def create_query_result(
```

Comment:
nit: internal functions should be named with prefix `_` (e.g. `_create_query_result`)

Reply:
Same applies to other `WorksheetQueryResultService` methods: `build_s3_file_path`, `convert_csv_to_xlsx`

Reply:
changes are done